code/_onclick/hud/alert.dm

```diff
@@ -0,0 +1,443 @@
+//A system to manage and display alerts on screen without needing you to do it yourself
+
+//PUBLIC -  call these wherever you want
+
+
+/mob/proc/throw_alert(category, type, severity, obj/new_master)
+
+/* Proc to create or update an alert. Returns the alert if the alert is new or updated, 0 if it was thrown already
+ category is a text string. Each mob may only have one alert per category; the previous one will be replaced
+ path is a type path of the actual alert type to throw
+ severity is an optional number that will be placed at the end of the icon_state for this alert
+ For example, high pressure's icon_state is "highpressure" and can be serverity 1 or 2 to get "highpressure1" or "highpressure2"
+ new_master is optional and sets the alert's icon state to "template" in the ui_style icons with the master as an overlay.
+ Clicks are forwarded to master */
+
+	if(!category)
+		return
+
+	var/obj/screen/alert/alert
+	if(alerts[category])
+		alert = alerts[category]
+		if(new_master && new_master != alert.master)
+			WARNING("[src] threw alert [category] with new_master [new_master] while already having that alert with master [alert.master]")
+			clear_alert(category)
+			return .()
+		else if(alert.type != type)
+			clear_alert(category)
+			return .()
+		else if(!severity || severity == alert.severity)
+			if(alert.timeout)
+				clear_alert(category)
+				return .()
+			else //no need to update
+				return 0
+	else
+		alert = new type
+
+	if(new_master)
+		var/old_layer = new_master.layer
+		var/old_plane = new_master.plane
+		new_master.layer = LAYER_HUD_ABOVE
+		new_master.plane = PLANE_PLAYER_HUD_ABOVE
+		alert.overlays += new_master
+		new_master.layer = old_layer
+		new_master.plane = old_plane
+		alert.icon_state = "template" // We'll set the icon to the client's ui pref in reorganize_alerts()
+		alert.master = new_master
+	else
+		alert.icon_state = "[initial(alert.icon_state)][severity]"
+		alert.severity = severity
+
+	alerts[category] = alert
+	if(client && hud_used)
+		hud_used.reorganize_alerts()
+	alert.transform = matrix(32, 6, MATRIX_TRANSLATE)
+	animate(alert, transform = matrix(), time = 2.5, easing = CUBIC_EASING)
+
+	if(alert.timeout)
+		spawn(alert.timeout)
```

Comment:
Could this use timers to callback clear_alert()?

Reply:
This has been done, and compile worked successfully. Code was used from here and edited slightly: https://github.com/tgstation/tgstation/blob/master/code/_onclick/hud/alert.dm#L67-L74